ledger/apply/keyreg.go

```diff
@@ -45,13 +46,15 @@ func Keyreg(keyreg transactions.KeyregTxnFields, header transactions.Header, bal
 		return fmt.Errorf("cannot change online/offline status of non-participating account %v", header.Sender)
 	}
 
+	params := balances.ConsensusParams()
+
 	// Update the registered keys and mark account as online
 	// (or, if the voting or selection keys are zero, offline/not-participating)
 	record.VoteID = keyreg.VotePK
 	record.SelectionID = keyreg.SelectionPK
```

Comment:
missing
```
record.StateProofID = keyreg.StateProofPK
```

Reply:
yes in my PR I had this up here (wrapped in the config check) why is it in two places here? 
https://github.com/algorand/go-algorand/pull/4092/files#diff-bed1aa7f95a52f456332f49968bd26e923ec08a7c69761de9396c070d6b048ae